### PR TITLE
Shuffling teams - equal team limit in each group

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

**Review by tknowak — APPROVED** [review]

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java [hunk 1]
```diff
@@ -53,13 +53,20 @@ public void createChampionship(ChampionshipDTO championshipDTO) {
     public void shuffleTeams(long champId) {
         List<RegisterEntity> registeredTeamList = registerRepository.findByChampId(champId);
         Random random = new Random();
+        int counterA =0;
+        int counterB =0;
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipService.java]: (nice to have) please use ctrl+alt+L to fix styling, like adding spaces after =
```suggestion
        int counterA = 0;
        int counterB = 0;
```

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java [hunk 2]
```diff
@@ -53,13 +53,20 @@ public void createChampionship(ChampionshipDTO championshipDTO) {
...
+
         for (RegisterEntity registeredTeamsGroup : registeredTeamList) {
             double randomNumber = random.nextInt(100);
             String group;
-            if (randomNumber >= 50) {
+            int countedTeams = registeredTeamList.size();
+            int maxTeamsInGroup = countedTeams / 2;
+            if (counterB >= maxTeamsInGroup || randomNumber >= 50 && counterA < maxTeamsInGroup) {
                 group = "A";
+                counterA += 1;
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipService.java]:
```suggestion
                counterA++;
```

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java [hunk 3]
```diff
@@ -53,13 +53,20 @@ public void createChampionship(ChampionshipDTO championshipDTO) {
...
             } else {
                 group = "B";
+                counterB += 1;
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipService.java]:
```suggestion
                counterB++;
```

**Review by kbnowak — APPROVED** [review]

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java (hunk 3 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/championships/ChampionshipService.java]: Fixed

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java (hunk 2 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/championships/ChampionshipService.java]: Fixed

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java (hunk 1 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/championships/ChampionshipService.java]: Fixed

**Pull request merged**